wasmer/instance.go

```diff
@@ -152,7 +152,7 @@ func NewInstanceWithOptions(
 	}
 
 	instance, err := newInstance(c_instance)
-	if instance.Memory != nil {
+	if instance != nil && instance.Memory != nil {
```

Comment:
no need for early exit- error is returned and checked if there is error when starting arwen. These thinks has extremely low probability to happen. This can happen in machines on which wasmer does not work.